sktime/detection/base/_base.py

```diff
@@ -962,8 +965,37 @@ def segments_to_change_points(y_sparse):
         2    7
         dtype: int64
         """
+        # Check if y_sparse is a numpy array and convert to pandas Series
+        if isinstance(y_sparse, np.ndarray):
```

Comment:
As a group, we haven't yet defined what an array of segments looks like so I would prefer to leave this out of this MR.

What do you propose an array of segments should look like? If you like we can always open another issue for this specific feature so we can input from others.

Reply:
Well I am not sure if we need to define explicitly what the **array of segments** would look like!
But on its counterpart, I have converted the `change_points` to a `pd.Series`, as visible below:

![image](https://github.com/user-attachments/assets/5b716737-58f9-4f7b-b547-637aba2cfb23)


This change is resulting all the **failing tests** to pass for the detector, here is the output:

![image](https://github.com/user-attachments/assets/d12b243d-ca5a-492e-ac1a-d2e41d24a93c)

@Alex-JG3 @fkiraly 
Could we proceed with this methodology?

Reply:
I don't think the current test suite in `TestAllDetectors` will cover passing a numpy array into `segments_to_change_points`. So passing those tests may give us a false sense of security. 

There is a test for `segments_to_change_points` called `test_segments_to_change_points` which you can find here:

https://github.com/sktime/sktime/blob/cc1f7bdc20fecfd91aad5339b88fa8ae4d463109/sktime/detection/tests/test_base.py#L137

This test only tests sparse segments as pandas series. For clarity, here is a code example showing what that series of sparse segments looks like.

```python
>>> import pandas as pd
>>> sparse_segments = pd.Series(
...     [1, -1, 2],
...     index=pd.IntervalIndex.from_breaks([2, 5, 7, 9], closed="left"),
... )
>>> print(sparse_segments)
[2, 5)    1
[5, 7)   -1
[7, 9)    2
dtype: int64
```

In order to support numpy array sparse segments, we need to decide the best way to represent example above as a numpy array. One way we could do this is using a 2D array where the first two columns the start and end points of the segments and the final column is the label of the change point. So the example above would be:

```
[
    [2, 5, 1],
    [5, 7, -1],
    [7, 9, 2]
]
```

As I understand, your approach looks for the index positions of the maximum and minimum values of `y_sparse`. Could this be affected by the `-1` in the 2D array example above?

This is a valuable discussion to have but I am concious is out of scope for this PR and I know there are several others who will have opinions on this. Would you consider opening a separate issues for that references this PR for better visibility with the rest of the team?

Reply:
@Alex-JG3 
Yeah! Sounds reasonable
Ok then I shall open an issue regarding this.